lib/services/remote_sync_service.dart

```diff
@@ -268,6 +270,8 @@ class RemoteSyncService {
 
     if (toBeUploaded > 0) {
       Bus.instance.fire(SyncStatusUpdate(SyncStatus.preparing_for_upload));
+      // verify if the files can b uploaded before initiating actual upload.
+      await _uploader.canUpload(toBeUploaded);
```

Comment:
`canUpload` doesn't do anything, so we could just expose `fetchURLs`, call that here with a comment on why we are executing it here?